src/models/task.rs

```diff
@@ -5,31 +5,40 @@ use chrono::NaiveDateTime;
 use serde::{Deserialize, Serialize};
 use std::cmp::Ordering;
 
-/// Tasks are generated from Goals, and represent a concrete activity of a
-/// specified duration, that is to be carried out at a specific time.
-/// A task can be in a number of different Task Statuses.  
+/// Tasks/Increments are generated to achieve a Goal in one or more Increments.
 /// A Goal can generate one or more Tasks.
-/// # Examples (greatly simplified to highlight the concept)
-/// | User Enters | Goal | Task(s) |
-/// | ----------- | ---- | ------- |
-/// | 'Dentist 1hr after 10 before 11' | Goal {..., after_time: 10, before_time: 11} | Task {..., id: 1, confirmed_start: 2023-01-23T10:00:00 , confirmed_deadline: 2023-01-23T11:00:00} |
-/// | 'Read 1hr daily' | Goal {..., repeat: daily} | Task {..., id: 1, confirmed_start: 2023-01-23T17:00:00, confirmed_deadline: 2023-01-23T18:00:00 }, Task {..., id: 2, confirmed_start: 2023-01-24T17:00:00, confirmed_deadline: 2024-01-24T18:00:00 }  
 #[derive(Deserialize, Debug, Eq, Clone)]
 pub struct Task {
+    /// Only used by the scheduler.
+    /// Unstable between scheduler runs if input changes.
     pub id: usize,
+    /// Reference to the Goal a Taks/Increment was generated from.
     pub goal_id: String,
+    /// Title of the Goal the Task/Increment was generated from.
+    /// Duplicated for ease of debugging and simplicity of code.
     pub title: String,
+    /// Duration the Task/Increment wants to claim on the Calendar.
+    /// This duration is equal or part of the Goal duration.
     pub duration: usize,
+    /// Used for finding next Task/Increment to be scheduled in combination with Task/Increment flexibility, after_goals and Tags.
     pub status: TaskStatus,
+    /// Used for finding next Task/Increment to be scheduled in combination with Task/Increment Status, after_goals and Tags.
     pub flexibility: usize,
+    /// Final start time for Task/Increment on Calendar - should be removed in favor of Timeline + SlotStatus combination.
     pub start: Option<NaiveDateTime>,
+    /// Final end time for Task/Increment on Calendar - should be removed in favor of Timeline + SlotStatus combination.
     pub deadline: Option<NaiveDateTime>,
+    /// The places on Calendar that could potentially be used given the Goal constraints - and what other scheduled Tasks/Increments already have consumed.
     pub slots: Vec<Slot>,
+    /// Used for finding next Task/Increment to be scheduled in combination with Task/Increment flexibility, after_goals, and Status.
```

Comment:
Is that the definition of Tags, which it seems the same definition of `after_goals`!

Reply:
Fixed!